### PR TITLE
Support additionalMetadata on an instance

### PR DESCRIPTION
**Pull request by justinsb**
Because of the limitations of labels & tags, metadata is often used to
store tracking information.

/kind feature
/kind api-change
```release-note
Support additionalMetadata on GCPMachine and GCPMachineTemplate, for
custom metadata fields.
```

### CODE REVIEW

**Comment by justinsb** [comment]
This is a revamp of #186 that I lost track of.

@vincepri previously asked whether we should do `map[string]string` instead; I don't have a strong opinion, but my preference is for this mode because it better reflects the underlying API (which in theory supports null values) and also because it can support additional modes / fields (e.g. if we wanted to get a value from secret or configmap).

**Review by cpanato — APPROVED** [review]
/lgtm

thanks @justinsb

File: api/v1alpha3/gcpmachine_types.go [hunk 1]
```diff
@@ -64,6 +64,11 @@ type GCPMachineSpec struct {
 	// +optional
 	AdditionalLabels Labels `json:"additionalLabels,omitempty"`
 
+	// AdditionalMetadata is an optional set of metadata to add to an instance, in addition to the ones added by default by the
+	// GCP provider.
+	// +optional
+	AdditionalMetadata []MetadataItem `json:"additionalMetadata,omitempty"`
```
Comment (detiber) [inline api/v1alpha3/gcpmachine_types.go:70]: I think this should probably also have some markers set to indicate that additionalMetadata should be treated as type `x-kubernetes-list-type` of `map`:
```suggestion
	// AdditionalMetadata is an optional set of metadata to add to an instance, in addition to the ones added by default by the
	// GCP provider.
	// +listType=map
	// +listMapKey=key
	// +optional
	AdditionalMetadata []MetadataItem `json:"additionalMetadata,omitempty"`
```

Comment (cpanato) [reply api/v1alpha3/gcpmachine_types.go:70]: @justinsb can you take a look at Jason's comment when you have a chance?

Comment (justinsb) [reply api/v1alpha3/gcpmachine_types.go]: Done - great point @detiber  - and thanks for the ping @cpanato !

**Review by cpanato — APPROVED** [review]
/lgtm

**Pull request merged**